### DISCUSSION TITLE
Google Analytics - How to push custom dimensions before page view?

### DISCUSSION THREAD

**Discussion by ronenteva** [nuxt/scripts#226] [Q&A]
```
const {dataLayer} = useScriptGoogleAnalytics();
dataLayer.push({content_group: 'shirts'})
```
If I understand correctly, the content_group will be sent with the next pageview.
What's the correct way of setting custom dimensions?

**Comment by harlan-zw** [comment]
Hi, to confirm are you trying to do something like this?

```ts
const { proxy } = useScriptGoogleAnalytics();

proxy.gtag('config', 'reset')
proxy.gtag('config', 'UA-XXXXXXX-1', { 'content_group': 'shirts' });
```

**Reply by ronenteva** [reply]
Yes, thank you

**Reply by harlan-zw** [reply]
No problem, ideally you'd be able to do this in the registry signature.

Let me know if you have any issues with it.

cc: @huang-julien, @flashdesignory

**Reply by ronenteva** [reply]
Oh sorry, I didn't realize that's an actual solution.

Is there really a `gtag('config', 'reset')` that I'm not aware of?
Won't this send two pageviews, one before the reset and one after?

**Comment by harlan-zw** [comment]
Closing as answered — `proxy.gtag('config', '<id>', { content_group: 'shirts' })` before the next page_view is the documented pattern. Reopen / new issue if you hit a specific edge case 🙏
